internal/subject/database/repository.go

```diff
@@ -1,84 +1,116 @@
 package database
 
 import (
+	"fmt"
+
 	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
 
 	"github.com/sumelms/microservice-course/internal/subject/domain"
 	"github.com/sumelms/microservice-course/pkg/errors"
 )
 
-type Repository struct {
-	*sqlx.DB
+const (
+	listSubject    = "list subject"
+	getSubject     = "get subject by uuid"
+	delecteSubject = "delete subject by uuid"
+)
+
+func NewRepository(db *sqlx.DB) (repository, error) {
+	sqlStatements := make(map[string]*sqlx.Stmt)
+
+	for queryName, query := range queries() {
+		stmt, err := db.Preparex(string(query))
+		if err != nil {
+			return repository{}, errors.WrapErrorf(err, errors.ErrCodeUnknown, "error preparing statement %s", queryName)
+		}
+		sqlStatements[queryName] = stmt
+	}
+	fmt.Println(sqlStatements)
+	return repository{
+		statements: sqlStatements,
+	}, nil
+}
+
+type repository struct {
```

Comment:
It needs to be exported, 'cause the `NewRepository` returns its type.

Reply:
I see. Agreed. In this case, we also have the interface that is exported from the domain package and can be used. It makes totally sense. I think that reporting the `unexported-return` from the linting tool needs to be disabled.